dydx3/starkex/starkex_resources/signature.py

```diff
@@ -174,6 +194,11 @@ def mimic_ec_mult_air(m: int, point: ECPoint, shift_point: ECPoint) -> ECPoint:
 
 
 def verify(msg_hash: int, r: int, s: int, public_key: Union[int, ECPoint]) -> bool:
+    if CPP_LIB_PATH is not None:
+        w = inv_mod_curve_size(s)
+        inverse_is_valid = cpp_verify(msg_hash=msg_hash, r=r, s=w, stark_key=public_key)
+        return inverse_is_valid or cpp_verify(msg_hash=msg_hash, r=r, s=s, stark_key=public_key)
```

Comment:
I misled you a little, instead of testing both cases we can assume the cpp library always uses the inverse of python.

We should invert the `s` value returned by the cpp sign function. And then we should invert `s` here before calling cpp verify. Then the Python verify code path should not have to change (should only do a single check, not two).